app/extensions/__init__.py

```diff
@@ -555,7 +562,8 @@ def set_custom_field_values(self, set_dict):
                 raise ValueError(
                     f'value {value} not valid for customField definition id {cfd_id} (on {self})'
                 )
-            cf[cfd_id] = value
+            defn = SiteSettingCustomFields.get_definition(self.__class__.__name__, cfd_id)
```

Comment:
Would it make sense to have a helper within the SiteSettingsCustomFileds that did this as you seem to be calling it a couple of times which is my cutoff ::)

Reply:
you mean `get_definition()` + `serialize_value()` ?